app/models/scheduled_event.rb

```diff
@@ -1,12 +1,62 @@
 class ScheduledEvent
   include Mongoid::Document
+  include Mongoid::Timestamps
 
   field :type, type: String
   field :event_name, type: String
-  field :start_date, type: Date
-  field :one_time, type: Boolean, default: false
-  field :recurring_rules, type: String
+  field :start_time, type: Date
+  field :one_time, type: Boolean, default: true
+  field :recurring_rules, type: Hash
   field :offset_rule, type: Integer, default: none
 
-  validates_presence_of :type, :event_name, :one_time, :start_date
+  embeds_many :event_exceptions
+
+  validates_presence_of :type, :event_name, :one_time, :start_time, :offset_rule
+
+  EVENT_TYPES = %W(holiday system_event)
+  HOLIDAYS = %W(New_Year MartinLuthor_birthdday washingtons_day memorial_day independence_day
```

Comment:
can we fix the inconsistent undercasing / capitalization ?

new_year, martin_luther_king_birthday , etc .